src/OrchardCore.Modules/OrchardCore.Localization/CulturePickerShape.cs

```diff
@@ -0,0 +1,36 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using OrchardCore.DisplayManagement;
+using OrchardCore.DisplayManagement.Descriptors;
+
+namespace OrchardCore.Menu
+{
+    public class CulturePickerShapes : IShapeTableProvider
+    {
+        public void Discover(ShapeTableBuilder builder)
+        {
+            builder.Describe("CulturePicker")
+                .OnProcessing(async context =>
+                {
+                    var culturePicker = context.Shape;
+                    var shapeFactory = context.ServiceProvider.GetRequiredService<IShapeFactory>();
+                    var httpContextAccessor = context.ServiceProvider.GetRequiredService<IHttpContextAccessor>();
+                    var httpContext = httpContextAccessor.HttpContext;
+                    var localizationOptions = httpContext.RequestServices.GetService<IOptions<RequestLocalizationOptions>>();
+
```

Comment:
Not required anymore.

Reply:
I'm not able to set the model from `Discover ()` waiting for help to move almost the code from the view to here